### ISSUE TITLE
multiple stylesheets for one tileset (instead of vice versa)

### ISSUE THREAD

**Issue by roelarents** [opengeospatial/ogcapi-tiles#111]
tl;dr: how to properly serve multiple styles for one tileset?

If I understand the docs correctly, when I'd like to serve styles for my (vector) tiles, I should use the `.dataset.style.vector.getTile*` operations, which seem a combination of the tiles and styles APIs. 

```
/{dataset and optionally a collection}/styles/{styleId}/tiles/{tileMatrixSetId}/{tileMatrix}/{tileRow}/{tileCol}
^ common                              ^ styles api    ^ tiles api
```

That implies a one-to-many relationship from style to tilesets. However, my/our use case is the other way around: one tileset which can be visualized by multiple styles. E.g. use a topographical map usable as a background (grey shades) or foreground layer (colorful). I'm wondering how to do this properly. Options:

1. Serve the same tiles with different styles on different endpoints.  
  `/styles/styleA/tiles/someMatrix/0/0/0` and  `styles/styleB/tiles/someMatrix/0/0/0` would yield the same exact tile. Not a big problem, but there would be no canonical url for the tile(set). Perhaps serve a 301/302 to `/tiles/someMatrix/0/0/0`.
2. Serve the tiles and styles seperately.  
  tiles via `/tiles`  
  nothing on `/styles/styleA/tiles`  
  styles on `/styles/styleA` and `/styles/styleB`
  `styles/styleA` which would yield a mapbox stylesheet includes the above tiles url (+x/y/z) as the source url  
  `styles/styleA/metadata` maybe also includes a link to the tileset.
3. Put the styles in the [tileset metadata](https://github.com/opengeospatial/ogcapi-tiles/blob/master/openapi/schemas/tms/tileSet.yaml#L63). But I don't think it's supposed to go there, plus there's only room for one.

I think option 2. But I'm not sure that's the way it's intended / allowed.

**Comment by jerstlouis** [comment]
@roelarents

Short answers: 2, except for styles with filtering rules, then 1. See also [GeoDataClass](https://github.com/opengeospatial/NamingAuthority/issues/97) for indicating styles / tilesets / collection compatibility.

> However, my/our use case is the other way around: one tileset which can be visualized by multiple styles. 

**Vector** tilesets can be rendered by a client using multiple styles. In this case, the tilesets should be available outside of a particular style path, i.e. `/tiles` for dataset tiles and `/collections/{collectionId}/tiles` for tiles of a particular collection.

However, the filtering aspect of styling may also remove some features so that it contains less data, especially for point features which cannot be generalized based only on scale. This is the use case for serving vector tilesets for a particular style (i.e. `/styles/{styleId}/tiles` at the root of the API), as the tiles will already have been pre-filtered and will only be suitable for that particular style (or one with identical filtering rules).

Even if the vector tilesets are available from a single location, multiple styles can be available and served by the API which will work for the same tileset, and rendered **map** tilesets of that same data (rendered server-side or pre-rendered) can be available for multiple styles at `/styles/{styleId}/map/tiles` or `/collections/{collectionId}/styles/{styleId}/map/tiles`.

In terms of compatibility between styles and tilesets, the concept of a [GeoDataClass](https://github.com/opengeospatial/NamingAuthority/issues/97) has been proposed as a URI that both the collections / tilesets and the styles can ascribe to. The URI would also point to a resource where the schemas corresponding to the GeoDataClass where available properties / fields etc. are described. So both the styles being served and the tilesets being served could use the same GeoDataClass URI and this way indicate to the clients that they can be used together.

See also the [geodataclasses](https://github.com/opengeospatial/ogcapi-tiles/blob/master/openapi/schemas/tms/geospatialData.yaml#L83) field in the [TileSet metadata](https://raw.githubusercontent.com/opengeospatial/2D-Tile-Matrix-Set/master/standard/figures/TileSet.png)'s GeospatialData class.
